### ISSUE TITLE
Autoradioctrl locale support

### ISSUE THREAD

**Issue by eineki** [pat1/autoradio#24]
Autoradioctrl --syncdb fails if the locale it_IT is not installed on the host machine.

Expected behaviour: autoradio should use any of the locale installed on the host

Workaround: install the italian language pack(s).

In debian/ubuntu : language-pack-it, language-pack-it-base

**Comment by eineki** [comment]
To reproduce the behaviour:
Launch ubuntustudio 16.04lts live iso with english language and italian keyboard. 
Install the autoradio prerequisite packages.
Download the latest autoradio zip file from github.
Follow the easy way install instructions:
    python setup.py build
    ./autoradioctrl --syncdb

**Comment by pat1** [comment]
./autoradioctrl --syncdb fail with some messages ?

**Comment by eineki** [comment]
Yes, full traceback:

    Traceback (most recent call last):
      File "./autoradioctrl", line 25, in <module>
        import autoradio.autoradio_config
      File "/home/ubuntu-studio/install/autoradio-master/autoradio/autoradio_config.py", line 75, in <module>
        locale.setlocale(locale.LC_ALL, config['autoradiod']['locale'])
      File "/usr/lib/python2.7/locale.py", line 581, in setlocale
        return _setlocale(category, locale)
    locale.Error: unsupported locale setting

**Comment by eineki** [comment]
A first patch could be (__./autoradio/autoradio_config.py__): 

    6a7
    > from warnings import warn
    75,76c76,83
    < locale.setlocale(locale.LC_ALL, config['autoradiod']['locale'])
    < 
    ---
    > try:
    >     locale.setlocale(locale.LC_ALL, config['autoradiod']['locale'])
    > except locale.Error:
    >     # unsupported locale?
    >     current_locale = '.'.join(locale.getdefaultlocale())
    >     warning_message = 'Unsupported locale: "%s" using "%s" as fallback' % (config['autoradiod']['locale'], current_locale)
    >     warn(warning_message)
    >     locale.setlocale(locale.LC_ALL, current_locale)

issue a warning and do a falback to the default locale for the host.
A patch with a silent fallback:

    75,76c75,80
    < locale.setlocale(locale.LC_ALL, config['autoradiod']['locale'])
    < 
    ---
    > try:
    >     locale.setlocale(locale.LC_ALL, config['autoradiod']['locale'])
    > except locale.Error:
    >     # unsupported locale?
    >     current_locale = '.'.join(locale.getdefaultlocale())
    >     locale.setlocale(locale.LC_ALL, current_locale)

**Comment by pat1** [comment]
@eineki  ti ho aggiunto come collaboratore puoi applicare la patch

**Comment by eineki** [comment]
Tentative patch:
The previous one did not consider the case in which the locale is not set in the host machine.

As stated into the locale python module:

>         
>       According to POSIX, a program which has not called
>        setlocale(LC_ALL, "") runs using the portable 'C' locale.
>        Calling setlocale(LC_ALL, "") lets it use the default locale as
>        defined by the LANG variable. Since we don't want to interfere
>        with the current locale setting we thus emulate the behavior
>        in the way described above.
>        

---

```
--- a/autoradio/autoradio_config.py
+++ b/autoradio/autoradio_config.py
@@ -4,6 +4,8 @@
 import os
 from configobj import ConfigObj,flatten_errors
 from validate import Validator
+import warnings
+

 configspec={}

@@ -72,5 +74,16 @@ group         = config['autoradiod']['group']
 env           = config['autoradiod']['env']

 import locale
-locale.setlocale(locale.LC_ALL, config['autoradiod']['locale'])
+try:
+    locale.setlocale(locale.LC_ALL, config['autoradiod']['locale'])
+except locale.Error:
+    # unsupported locale?
+    current_locale = locale.getdefaultlocale()
+    if current_locale[0] is None:
+        current_locale = ""
+    else:
+        current_locale = '.'.join(current_locale)
+    warning_message = 'Unsupported locale: "%s" using "%s" as fallback' % (config['autoradiod']['locale'], current_locale)
+    warnings.warn(warning_message)
+    locale.setlocale(locale.LC_ALL, current_locale)

**Comment by pat1** [comment]
closed by f94ff18db5e997f02578a11b7e12a558434a7a30
